modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -560,6 +560,11 @@ private void patchProperty(CodegenModel model, CodegenProperty property){
             property.datatypeWithEnum = property.datatypeWithEnum.replace("Dictionary>", property.items.datatypeWithEnum + ">");
             property.dataType = property.datatypeWithEnum;
         }
+
+        // HOTFIX: https://github.com/OpenAPITools/openapi-generator/issues/14944
+        if (property.datatypeWithEnum.equals("decimal")) {
+            property.isDecimal = true;
+        }
```

Comment:
looks like something maybe wrong in the default codegen. i'll take a look later and add a test if needed.